### DISCUSSION TITLE
Add support for viewing unread chats from first unread message

### DISCUSSION THREAD

**Discussion by arcxne** [d99kris/nchat#169] [Ideas]
In the official WhatsApp clients, when you open an unread chat, it shows the first unread message.
However, on nchat, it shows the most recent message. This is inconvenient as sometimes a chat might have many, many messages and it may be hard to go all the way up to see where all the unread messages start, especially to see if there is any important information.

**Comment by dvragulin** [comment]
Yes, I ran into the same problem on Telegram. You simply skip any discussion in groups at the moment because the last message opens.

There may be a way to save the current position of the read messages.

This problem is really critical, which makes it difficult to use nchat as a regular customer.

**Comment by d99kris** [comment]
Hi @arcxne - thanks for reporting a feature request. I think it's a reasonable feature request, but unfortunately it requires fairly large changes in how nchat manages its message retrieval, so there's no quick fix for this unfortunately. I will leave the request here and hopefully get to it some day, but not soon.

**Comment by n4ch0m4n** [comment]
Hi, same problem here, on Telegram!
thanks!

**Comment by desgua** [comment]
I made a pull request implementing that. The changes can be seen here:
https://github.com/d99kris/nchat/compare/master...desgua:nchat:add_unread_chat_reverse_order

**Reply by d99kris** [reply]
> I made a pull request implementing that. The changes can be seen here: [master...desgua:nchat:add_unread_chat_reverse_order](https://github.com/d99kris/nchat/compare/master...desgua:nchat:add_unread_chat_reverse_order)

I believe this feature request is proposing something different, namely to view messages (in message history view) from oldest unread message. It is not trivial to implement as some early design decision was made to fetch newest message(s) first, and lazily fetch older ones. So supporting this feature request would involve quite a bit of changes how messages are cached/etc, not "just" a ui change.

**Reply by desgua** [reply]
I finally understood this request. Thank you so much!
